modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/visualization_modes/VisualizationModes.tsx

```diff
@@ -0,0 +1,93 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import ClayLayout from '@clayui/layout';
+import ClayTabs from '@clayui/tabs';
+import React, {ComponentType, useState} from 'react';
+
+import {IFDSViewSectionProps} from '../../FDSView';
+import ListVisualizationMode, {
+	IListVisualizationMode,
+} from './list_visualization_mode/ListVisualizationMode';
+import {
+	ITableVisualizationMode,
+	TableVisualizationMode,
+} from './table_visualization_mode/TableVisualizationMode';
+
+export interface IBaseVisualizationMode<Type extends string> {
+	label: string;
+	type: Type;
+	visualizationModeId: string;
+}
+
+type IVisualizationMode = ITableVisualizationMode | IListVisualizationMode;
+
+const DEFAULT_VISUALIZATION_MODES: IVisualizationMode[] = [
+	{
+		label: Liferay.Language.get('table'),
+		type: 'table',
+		visualizationModeId: 'defaultTable',
+	},
+	{
+		label: Liferay.Language.get('list'),
+		type: 'list',
+		visualizationModeId: 'defaultList',
+	},
+];
+
+const VISUALIZATION_MODE_COMPONENT_MAP: {
+	[key in IVisualizationMode['type']]: ComponentType<IFDSViewSectionProps>;
+} = {
+	list: ListVisualizationMode,
+	table: TableVisualizationMode,
+};
```

Comment:
Can we add this to base interface? Something like 
```js
export interface IBaseVisualizationMode<Type extends string> {
    component: ComponentType<IFDSViewSectionProps>;
	label: string;
	type: Type;
	visualizationModeId: string;
}
```

Reply:
I am not sure about this. I wanted to keep the `IBaseVisualizationMode` as something that we can serialize and send to backend, in case that we need to persist that list (ex. if we implement the tab reordering).

That's why I didn't add the `component` to the list. If we do so we do not need the `type` attribute at all, just use the given component.

I've you're still convinced I can refactor that, no problem :smile:

Reply:
Hmm interesting point, I didn't consider serialization at all. We had some discussions on how to handle form serializations (cc @bryceosterhaus ) , as our method of [submitting in DSM](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/FDSEntries.tsx#L347-L361) is pretty manual, unstructured, and inconsistent as a result. An enforcement of form structure through an interface, while reusing it to build UI, seems to me a really neat idea! There a quite a few placeholders currently in PR, I'm curious to see exactly what you have in mind. Yeah, as far as I'm concerned, we can definitely leave it as it is. :+1:

Reply:
Alright, I'll leave it then. I don't have a clear idea of how we should send that info (still learning how the Dataset is structured :sweat_smile:) but at least I wanted to have something that we can push into a JSON (like it's already being done in the FDSEntry)... so yep, that's it.

Apart from that, with the "default" list of visualization modes I was just trying to keep something that we can put on a `useState` hook if needed (or context, reducer, whatever), so we can live update when we implement reordering. I am not completely sure about this because I don't know if we are going to have some page reload when the settings are updated.

Reply:
The last page reload is when you to the View page, the one with all tabs. After that, it's all async requests and React updates on response.